geo/src/algorithm/interior_point.rs

```diff
@@ -0,0 +1,787 @@
+use std::cmp::Ordering;
+
+use crate::algorithm::{
+    bounding_rect::BoundingRect, centroid::Centroid, coords_iter::CoordsIter,
+    euclidean_distance::EuclideanDistance, line_intersection::LineIntersection,
+    lines_iter::LinesIter, relate::Relate,
+};
+use crate::geometry::*;
+use crate::sweep::*;
+use crate::GeoFloat;
+
+/// Calculation of interior points.
+/// An interior point is a point that's guaranteed to intersect a given geometry, and will be
+/// strictly on the interior of the geometry if possible, or on the edge of the geometry has zero
+/// area. A best effort will additionally be made to locate the point reasonably centrally.
+///
+/// For polygons, this point is located by drawing a line that approximately subdivides the
+/// bounding box around the polygon in half, intersecting it with the polygon, then calculating
+/// the midpoint of the longest line produced by the intersection. For lines, the non-endpoint
+/// vertex closest to the line's centroid is returned if the line has interior points, or an
+/// endpoint is returned otherwise.
+///
+/// For multi-geometries or collections, the interior points of the constituent components are
+/// calculated, and one of those is returned (for MultiPolygons, it's the point that's the midpoint
+/// of the longest intersection of the intersection lines of any of the constituent polygons, as
+/// described above; for all others, the interior point closest to the collection's centroid is
+/// used).
+///
+/// # Examples
+///
+/// ```
+/// use geo::InteriorPoint;
+/// use geo::{point, polygon};
+///
+/// // rhombus shaped polygon
+/// let polygon = polygon![
+///     (x: -2., y: 1.),
+///     (x: 1., y: 3.),
+///     (x: 4., y: 1.),
+///     (x: 1., y: -1.),
+///     (x: -2., y: 1.),
+/// ];
+///
+/// assert_eq!(
+///     Some(point!(x: 1., y: 2.)),
+///     polygon.interior_point(),
+/// );
+/// ```
+pub trait InteriorPoint {
+    type Output;
+
+    /// See: <https://en.wikipedia.org/wiki/InteriorPoint>
```

Comment:
This wiki page doesn't seem to exist. Is this aspirational? Or a broken link?

Reply:
Nope, my mistake. I started with the centroid codebase as a skeleton and did some up-front search-and-replace (and there *is* a centroid page). Just changed this to a prose description.